README.md

```diff
@@ -291,173 +321,204 @@ be used in global variable initializers.
 
-The maximum size for the UTF-8 encoding of an individual string literal
+The maximum size for the WTF-8 encoding of an individual string literal
 is 2<sup>31</sup>–1 bytes.  Embeddings may impose their own limits which
 are more restricted.  But similarly to `string.new`, instantiating a
 module with string literals may fail due to lack of memory resources,
 even if the string size is formally within the limits.  However
 `string.const` itself never traps when passed a valid literal offset.
 
-### String cursors
-
-To represent positions in strings for the purposes of accessing string
-contents, we introduce the concept of "cursor", encoded as an `i32`.  A
-string consisting of *N* unicode scalar values will have *N*+1 distinct
-valid cursor values: one before each USV, and one at the end.  A cursor
-value of 0 denotes the string start (before the first USV).  It follows
-that 0 may also denote the end of the string also, for zero-length
-strings.
-
-A cursor value is an offset into a string's contents.  Cursors uniquely
-identify a position in a string, and are ordered, and therefore can be
-compared against each other.
-
-If a host represents string contents internally using UTF-8, UTF-16, or
-UTF-32, or a variant thereof such as the one used in JavaScript, cursor
-values are code unit offsets.
-
-For example, because JavaScript hosts have to represent strings as
-(logical) sequences of 16-bit code units, a WebAssembly string cursor
-for a WebAssembly implementation embedded in a web browser will be a
-code unit offset (the same as the operand to JavaScript's
-[`String.prototype.charCodeAt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charCodeAt).
-
-For a WebAssembly implementation that represents strings as UTF-8
-internally, cursor values are byte offsets.  The intention is that
-accessing content in a string with a cursor has the least possible
-overhead.  This also allows hosts to communicate string positions with
-WebAssembly programs.
-
-If a host does not represent strings using a Unicode encoding scheme,
-the specific mapping from USV offset to cursor value is
-implementation-defined, with the requirement that values be ordered and
-that each position must have one and only one cursor value.
-
-To move a cursor to a new position, use the `string.advance` and
-`string.rewind` seek instructions.
-
-```
-(string.advance str:stringref cursor:i32 codepoints:i32)
-  -> cursor:i32, codepoints:i32
-(string.rewind str:stringref cursor:i32 codepoints:i32)
-  -> cursor:i32, codepoints:i32
-```
-Return a new cursor into the string *`str`* which is *`codepoints`*
-codepoints after or before *`cursor`*.  Also return the number of
-codepoints by which the new cursor differs from the old, which may be
-less than the *`codepoints`* parameter if end-of-string or
-beginning-of-string would be reached.  The *`codepoints`* argument is
-interpreted as a `u32`.
+### String positions
```

Comment:
I'm wondering whether we can get away with not having cursors/positions at all, and instead entirely rely on `string.encode_*`ing the string's contents to memory (or a WasmGC array). Source languages that must offer an implementation of `character-at` would compile this to the creation of a temporary copy of the string, on which they can then perform whatever encoding-specific cursor manipulations they want. If we assume that `character-at` is typically used for iterating over (most of) an entire string, then this explicit conversion is probably only slightly more expensive than the currently proposed position-based approach (especially considering position validity checking), so that might be acceptable at least for an MVP. Creating string slices would then also involve dumping the source string to memory and `string.new_*`ing the desired part back into a string; I don't have a good intuition for how frequent this is.
To be clear, I'm not trying to argue for this change, just putting the idea out there. I'm honestly not sure about its feasibility.

Reply:
I guess #31 would be this issue.  Thanks for filing that!